### PR TITLE
MNTOR-1930: add 3 breach types for easier rendering

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-1930

# Description
Add 3 more arrays for the "Action needed" UI

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/functions/server/getUserBreaches.ts [hunk 1]
```diff
@@ -36,6 +43,25 @@ export async function getUserBreaches({
...
+      }
+
+      if (b.DataClasses.includes(BreachDataTypes.Passwords)) {
+        passwordBreaches.push(b);
+      }
+
+      if (b.DataClasses.includes(BreachDataTypes.Phone)) {
+        phoneBreaches.push(b);
+      }
+    });
```
Comment (maxxcrawford) [inline src/app/functions/server/getUserBreaches.ts:62]: I was trying to come up with a more legible/DRY solution, but I think this works! 

My one edge case is if I have a breach that includes both SSN and Passwords, would that specific breach be in both arrays? If so, is that okay? (Question may be for @codemist)

File: src/app/functions/server/getUserBreaches.ts [hunk 2]
```diff
@@ -36,6 +43,25 @@ export async function getUserBreaches({
   const breachesData = await getAllEmailsAndBreaches(subscriber, allBreaches);
   appendBreachResolutionChecklist(breachesData, options);
 
+  const ssnBreaches: HibpLikeDbBreach[] = [];
+  const passwordBreaches: HibpLikeDbBreach[] = [];
+  const phoneBreaches: HibpLikeDbBreach[] = [];
+  for (const { breaches } of breachesData.verifiedEmails) {
+    breaches.forEach((b) => {
+      if (b.DataClasses.includes(BreachDataTypes.SSN)) {
```
Comment (maxxcrawford) [inline src/app/functions/server/getUserBreaches.ts:51]: TIL about the `BreachDataTypes`. Love it!

**Review by maxxcrawford — COMMENTED** [review]
No blockers, but wanna make sure the edge case I outlined works for @codemist

File: src/app/functions/server/getUserBreaches.ts (hunk 1 above)
Comment (mansaj) [reply src/app/functions/server/getUserBreaches.ts:62]: The assumption is that it would count as both, thus in both arrays.

**Review by flozia — APPROVED** [review]
Looks good!

**Pull request merged**